### PR TITLE
[RLlib] Cleanup `examples` folder 04: Curriculum and checkpoint-by-custom-criteria examples moved to new API stack.

### PR DESCRIPTION
**Pull request by sven1977**
Cleanup `examples` folder 04:
* Curriculum example moved to new API stack.
* checkpoint-by-custom-criteria example moved to new API stack.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/examples/curriculum/curriculum_learning.py [hunk 1]
```diff
@@ -1,141 +1,236 @@
...
         "timesteps_total": args.stop_timesteps,
-        "episode_reward_mean": args.stop_reward,
     }
 
-    tuner = tune.Tuner(
-        args.run,
-        param_space=config.to_dict(),
-        run_config=air.RunConfig(stop=stop, verbose=2),
+    run_rllib_example_script_experiment(
+        base_config, args, stop=stop, success_metric={"task_solved": 1.0}
```
Comment (simonsays1980) [inline rllib/examples/curriculum/curriculum_learning.py:235]: Very nice example :+1:

**Review by simonsays1980 — APPROVED** [review]
LGTM. Very happy about the curriculum example.

File: rllib/examples/checkpoints/checkpoint_by_custom_criteria.py [hunk 2]
```diff
@@ -1,67 +1,90 @@
...
-    choices=["tf", "tf2", "torch"],
-    default="torch",
-    help="The DL framework specifier.",
+How to run this script
+----------------------
+`python [script file name].py --enable-new-api-stack`
+
+For debugging, use the following additional command line options
+`--no-tune --num-env-runners=0`
+which should allow you to set breakpoints anywhere in the RLlib code and
```
[the remark this answers is not in the document]
Comment (sven1977) [reply rllib/examples/checkpoints/checkpoint_by_custom_criteria.py:22]: Absolutely! I'm always afraid, we are going to get rid of Ray local-mode at some point. Also, for any number of Learner workers > 0, local mode doesn't work (not sure why, actually).

File: rllib/examples/checkpoints/checkpoint_by_custom_criteria.py [hunk 3]
```diff
@@ -79,28 +102,32 @@
...
     # Checkpoint with the lowest policy loss value:
-    if config._enable_new_api_stack:
+    if args.enable_new_api_stack:
         policy_loss_key = "info/learner/default_policy/policy_loss"
     else:
         policy_loss_key = "info/learner/default_policy/learner_stats/policy_loss"
-    ckpt = results.get_best_result(metric=policy_loss_key, mode="min").checkpoint
-    print("Lowest pol-loss: {}".format(ckpt))
+    best_result = results.get_best_result(metric=policy_loss_key, mode="min")
+    ckpt = best_result.checkpoint
```
[the remark this answers is not in the document]
Comment (sven1977) [reply rllib/examples/checkpoints/checkpoint_by_custom_criteria.py:119]: Ah, cool, so `ckpt = best_result.checkpoint` returns the very last checkpoint only?

And if the last is not the best one, it's better to do:
`best_result.get_best_checkpoint(metric=policy_loss_key, mode="min")`
??

[the remark this answers is not in the document]
Comment (sven1977) [reply rllib/examples/checkpoints/checkpoint_by_custom_criteria.py:119]: This actually doesn't seem to work well with nested keys.
If I do `best_result.get_best_checkpoint(policy_loss_key, mode="min")`, I get:
```
RuntimeError: Invalid metric name ('info', 'learner', 'default_policy', 'learner_stats', 'policy_loss')! You may choose from the following metrics: dict_keys(['custom_metrics', 'episode_media', 'info', 'sampler_results', 'episode_reward_max', 'episode_reward_min', 'episode_reward_mean', 'episode_len_mean', 'episodes_this_iter', 'episodes_timesteps_total', 'policy_reward_min', 'policy_reward_max', 'policy_reward_mean', 'hist_stats', 'sampler_perf', 'num_faulty_episodes', 'connector_metrics', 'num_healthy_workers', 'num_in_flight_async_reqs', 'num_remote_worker_restarts', 'num_agent_steps_sampled', 'num_agent_steps_trained', 'num_env_steps_sampled', 'num_env_steps_trained', 'num_env_steps_sampled_this_iter', 'num_env_steps_trained_this_iter', 'num_env_steps_sampled_throughput_per_sec', 'num_env_steps_trained_throughput_per_sec', 'timesteps_total', 'num_steps_trained_this_iter', 'agent_timesteps_total', 'timers', 'counters', 'done', 'episodes_total', 'training_iteration', 'trial_id', 'date', 'timestamp', 'time_this_iter_s', 'time_total_s', 'pid', 'hostname', 'node_ip', 'config', 'time_since_restore', 'iterations_since_restore', 'perf', 'experiment_tag']).

```

**Pull request merged**